app/src/main/java/com/example/kuba10/mypokemonplaces/ListFragment/FirebaseListFragment.java

```diff
@@ -1,151 +0,0 @@
-
-
-public class FirebaseListFragment extends Fragment implements OnStartDragListener {
-
-
-    private AdapterForTouchAndFirebase mFirebaseAdapter;
-    private ItemTouchHelper mItemTouchHelper;
-    private FragmentListener fragmentListener;
-    private Query orderByChild;
-
-
-    @BindView(R.id.listRecycler)
-    RecyclerView recyclerView;
-
-
-    public static FirebaseListFragment newInstance() {
-        return new FirebaseListFragment();
-    }
-
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        fragmentListener = (FragmentListener) getActivity();
-
-    }
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.fragment_list, container, false);
-        ButterKnife.bind(this, view);
-
-
-        prepareQuery();
-        setUpFirebaseAdapter();
-
-        return view;
-    }
-
-    @Override
-    public void onStart() {
-        super.onStart();
-    }
-
-    private void setUpFirebaseAdapter() {
-
-        mFirebaseAdapter = new AdapterForTouchAndFirebase(PokePlace.class,
-                R.layout.place_card_layout, CardViewHolder.class,
-                orderByChild, this, this);
-
-        recyclerView.setHasFixedSize(true);
-        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
-        recyclerView.setLayoutManager(layoutManager);
-        recyclerView.setAdapter(mFirebaseAdapter);
-
-        ItemTouchHelper.Callback callback = new SimpleItemTouchHelperCallback(mFirebaseAdapter);
-        mItemTouchHelper = new ItemTouchHelper(callback);
-        mItemTouchHelper.attachToRecyclerView(recyclerView);
-    }
-
-    private void prepareQuery() {
-        orderByChild = FirebaseDatabase.getInstance()
-                .getReference()
-                .child(Constants.PLACES)
-                .orderByChild(Constants.FIREBASE_QUERY_INDEX);
-    }
-
-    @Override
-    public void onStartDrag(RecyclerView.ViewHolder viewHolder) {
-        mItemTouchHelper.startDrag(viewHolder);
-    }
-
-    @Override
-    public void onPause() {
-
-        super.onPause();
-        mFirebaseAdapter.cleanup();
-
-    }
-
-    public void dismiss() {
-
-        getActivity().getSupportFragmentManager()
-                .beginTransaction()
-                .setCustomAnimations(R.anim.enter, R.anim.exit, R.anim.pop_enter, R.anim.pop_exit)
-                .remove(this).commit();
-    }
-
-    public void findPosition(PokePlace place) {
-
-        fragmentListener.findPlaceFromList(place);
-
-
-    }
-
-
-    @Override
-    public void onDetach() {
-        super.onDetach();
-        fragmentListener = null;
-        orderByChild = null;
-        mItemTouchHelper = null;
-        mFirebaseAdapter = null;
-
-    }
-
-    public void refreshList(int size) {
-    }
-
-    public void openDetails(Fragment fragment) {
```

Comment:
Jeśli otwierasz już tworzysz metodę do otwarcia konkretnego fragmentu, nie przerzucaj w argumencie, tylko wszystko załatw w obrębie tej metody

Reply:
W ogóle to była nieużywana metoda - teraz adapter wywołuje dialogFragment z detalami. To w ogóle temat do rozmowy jak powinno wyglądać połaczenie fragmentu z adapterem - przerzucanie fragmentu jako "parentfragment" jest fajne bo daje dostep do kontekstu, metod, fragmentmanagera etc - ale jak rozumiem to zła metoda